app/assets/javascripts/host_power.js

```diff
@@ -0,0 +1,64 @@
+(function(){
+
+  $(document).ready(function () {
+    powerStatus();
+  });
+
+  function hostIds() {
+    var res = [];
+
+    $('#hosts-table tbody tr').each( function (index, item) {
+      res.push($(item).data("host-id"));
+    });
```

Comment:
Because this is what I get with `each` vs what I get with `map`:
![](http://i.imgur.com/4ufe2vh.png)